Sources/Sentry/SentryScope.m

```diff
@@ -154,6 +154,17 @@ - (void)setSpan:(nullable id<SentrySpan>)span
     }
 }
 
+- (void)setPropagationContext:(SentryPropagationContext *)propagationContext
+{
+    @synchronized(_propagationContext) {
+        _propagationContext = propagationContext;
+
+        for (id<SentryScopeObserver> observer in self.observers) {
+            [observer setTraceContext:[self.propagationContext traceContextForEvent]];
```

Comment:
`m`: `traceContextForEvent` is creating a new dictionary on every call. Can we reuse the same dictionary when sending it to observers to reduce allocations?

Reply:
Nice catch, thanks!